src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -1440,14 +1447,14 @@ private int MakeSeparatorList(ReadOnlySpan<char> separators, Span<int> sepList)
                             if (IsCharBitSet(charMap, (byte)c) && IsCharBitSet(charMap, (byte)(c >> 8)) &&
                                 separators.Contains(c))
                             {
-                                sepList[foundCount++] = i;
+                                sepListBuilder.Append(i);
                             }
                         }
                     }
                     break;
             }
 
-            return foundCount;
```

Comment:
Nit: I think it maybe a bit easier to understand it the `AsReadOnlySpan` call is done by the caller.